src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -149,15 +98,50 @@ private void consume() throws Exception {
         continue;
       }
 
-      if (record == null)
-        continue;
+      if (record == null) continue;
 
       GenericRecord avroRecord = Utils.genericRecordFromJson(record.value());
       if (avroRecord == null) {
         _sensors._consumeError.record();
         continue;
       }
       int partition = record.partition();
+
+      /* Commit availability and commit latency service */
+      try {
+        TopicPartition topicPartition = new TopicPartition(_topic, partition);
+        final AtomicBoolean callbackFired = new AtomicBoolean(false);
+        final AtomicReference<Exception> offsetCommitIssues = new AtomicReference<>(null);
+        OffsetAndMetadata offsetAndMetadata;
+        /* Call commitAsync, wait for a NON-NULL return value (see https://issues.apache.org/jira/browse/KAFKA-6183) */
+        OffsetCommitCallback commitCallback = new OffsetCommitCallback() {
+          @Override
+          public void onComplete(Map<TopicPartition, OffsetAndMetadata> topicPartitionOffsetAndMetadataMap, Exception exception) {
+            if (exception != null) {
+              offsetCommitIssues.set(exception);
+            }
+            callbackFired.set(true);
+          }
+        };
+
+        if (_offsetsToCommit != null && !_offsetsToCommit.isEmpty()) {
```

Comment:
Maybe don't commit offsets every message as it's kind of an extreme use case.  Perhaps select a time period, like every second or 5 seconds for  committing an offset.